bin/import_training_speech.py

```diff
@@ -0,0 +1,88 @@
+
+from os import path
+
+FIELDNAMES = ['wav_filename', 'wav_filesize', 'transcript']
+MAX_SECS = 10
+ARCHIVE_DIR_NAME = 'training_speech'
+ARCHIVE_NAME = '2018-09-27_fr_FR.zip'
+ARCHIVE_URL = 'https://s3.eu-west-3.amazonaws.com/audiocorp/releases/' + ARCHIVE_NAME
+
+SIMPLE_BAR = ['Progress ', progressbar.Bar(), ' ', progressbar.Percentage(), ' completed']
+
+def _download_and_preprocess_data(target_dir):
+    # Making path absolute
+    target_dir = path.abspath(target_dir)
+    # Conditionally download data
+    archive_path = _maybe_download(ARCHIVE_NAME, target_dir, ARCHIVE_URL)
+    # Conditionally extract common voice data
+    _maybe_extract(target_dir, ARCHIVE_DIR_NAME, archive_path)
+    # Conditionally convert common voice CSV files and mp3 data to DeepSpeech CSVs and wav
+    _maybe_convert_sets(target_dir, ARCHIVE_DIR_NAME)
+
+def _maybe_download(archive_name, target_dir, archive_url):
+    # If archive file does not exist, download it...
+    archive_path = path.join(target_dir, archive_name)
+    if not path.exists(archive_path):
+        print('No archive "%s" - downloading...' % archive_path)
+        req = requests.get(archive_url, stream=True)
+        total_size = int(req.headers.get('content-length', 0))
+        done = 0
+        with open(archive_path, 'wb') as f:
+            bar = progressbar.ProgressBar(max_value=total_size, widgets=SIMPLE_BAR)
+            for data in req.iter_content(1024*1024):
+                done += len(data)
+                f.write(data)
+                bar.update(done)
+    else:
+        print('Found archive "%s" - not downloading.' % archive_path)
+    return archive_path
+
+def _maybe_extract(target_dir, extracted_data, archive_path):
+    # If target_dir/extracted_data does not exist, extract archive in target_dir
+    extracted_path = path.join(target_dir, extracted_data)
+    if not path.exists(extracted_path):
+        print('No directory "%s" - extracting archive...' % extracted_path)
+        if not os.path.isdir(extracted_path):
+            os.mkdir(extracted_path)
+        with zipfile.ZipFile(archive_path) as zip_f:
+            zip_f.extractall(extracted_path)
+    else:
+        print('Found directory "%s" - not extracting it from archive.' % archive_path)
+
+def _maybe_convert_sets(target_dir, extracted_data):
+    extracted_dir = path.join(target_dir, extracted_data)
+    # override existing CSV with normalized one
+    target_csv = os.path.join(target_dir, 'training_speech_' + ARCHIVE_NAME.replace('.zip', '.csv'))
+    if os.path.isfile(target_csv):
+        return
+    path_to_original_csv = os.path.join(extracted_dir, 'data.csv')
+    with open(path_to_original_csv) as csv_f:
+        data = [
+            dict(
+                wav_filename=os.path.join(target_dir, extracted_data, d['path']),
+                wav_filesize=os.path.getsize(os.path.join(target_dir, extracted_data, d['path'])),
+                transcript=d['text'],  # TODO: cleanup (ex: rm punctuation)
```

Comment:
A cleanup is probably required here since transcript may contain punctuation (,.!? etc) and diatrics (éèêàç etc). Do you have an helper for that ?